### PR TITLE
Some structure generation and random stuff

### PR DESCRIPTION
**Pull request by Matrix89**
I don't think this should get merged now, I still need to deobf some names.

### CODE REVIEW

File: mappings/byy.mapping [hunk 1]
```diff
@@ -0,0 +1,7 @@
+CLASS byy
+	FIELD a x0 I
```
Comment (asiekierka) [inline mappings/byy.mapping]: minX/maxX/... is the convention

File: mappings/bzn.mapping [hunk 2]
```diff
@@ -0,0 +1,6 @@
+CLASS bzn
+	FIELD a logger Lorg/apache/logging/log4j/Logger;
+	FIELD d structurePart Ljava/util/Map;
+	FIELD e partStructure Ljava/util/Map;
+	METHOD a addPart (Ljava/lang/Class;Ljava/lang/String;)V
+	METHOD b registerStructure (Ljava/lang/Class;Ljava/lang/String;)V
```
Comment (asiekierka) [inline mappings/bzn.mapping:6]: Structure or Feature? (I'm not sure)

File: mappings/net/minecraft/sortme/structures/DesertTemple.mapping [hunk 3]
```diff
@@ -0,0 +1,6 @@
+CLASS bza net/minecraft/sortme/structures/DesertTemple
```
Comment (asiekierka) [inline mappings/net/minecraft/sortme/structures/DesertTemple.mapping]: DesertTempleStructure or StructureDesertTemple (again, no clear standard yet). not DesertTemple alone

**Review by asiekierka — CHANGES_REQUESTED** [review]

File: mappings/bzn.mapping (hunk 2 above)
Comment (Matrix89) [reply mappings/bzn.mapping:6]: These classes don't inherit form Feature only generate it, so more like FeatureGenerator

File: mappings/net/minecraft/sortme/structures/DesertTemple.mapping (hunk 3 above)
Comment (Matrix89) [reply mappings/net/minecraft/sortme/structures/DesertTemple.mapping]: Yep, named it only to keep track of it

File: mappings/bzn.mapping (hunk 2 above)
Comment (asiekierka) [reply mappings/bzn.mapping:6]: FeatureFactory?

File: mappings/net/minecraft/sortme/structures/DesertTemple.mapping (hunk 3 above)
Comment (asiekierka) [reply mappings/net/minecraft/sortme/structures/DesertTemple.mapping]: OK

File: mappings/bzn.mapping [hunk 4]
```diff
@@ -0,0 +1,6 @@
+CLASS bzn
+	FIELD a logger Lorg/apache/logging/log4j/Logger;
```
Comment (jamierocks) [inline mappings/bzn.mapping:2]: These fields are static, they should be upper-case.

```java
    private static final Logger a = LogManager.getLogger();
    private static final Map<String, Class<? extends bzr>> b = Maps.newHashMap();
    private static final Map<Class<? extends bzr>, String> c = Maps.newHashMap();
    private static final Map<String, Class<? extends bzq>> d = Maps.newHashMap();
    private static final Map<Class<? extends bzq>, String> e = Maps.newHashMap();
```

**Review by jamierocks — CHANGES_REQUESTED** [review]

**Comment by modmuss50** [comment]
Looks good, just need to fix that logger name.

I have a feeling this has a bit of overlap with #194

**Comment by asiekierka** [comment]
-> 18w46a

**Pull request closed without merging**